### DISCUSSION TITLE
Startable's close method may have Exception signature

### DISCUSSION THREAD

**Discussion by PMacho** [testcontainers/testcontainers-java#5044] [Ideas]
At the moment the `Startable` classes `close()` method looked like so:

`@Override
    default void close() {
        stop();
    }`

However, `Startable` implements `AutoClosable` where the close method has an Exception signature. I suggest to keep this signature in the `close()` method of `Starable`. 

Reason: It allows users to override the `close` method, executing other methods that might also throw exception, without explicitly catching them.

**Comment by bsideup** [comment]
This is done for convenience so that the **consumers** don't need to catch the exception, especially if they use `try-with-resources`. We generally prefer making this easier at the consumption side, not implementation, and a single `try-catch` in `close()` is much better than many `catch`es at use use :)
